include/CL/opencl.hpp

```diff
@@ -10253,6 +10369,482 @@ namespace compatibility {
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clRetainCommandBufferKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clReleaseCommandBufferKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clGetCommandBufferInfoKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clEnqueueCommandBufferKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCommandBarrierWithWaitListKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCommandCopyBufferKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCommandCopyBufferRectKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCommandCopyBufferToImageKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCommandCopyImageKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCommandCopyImageToBufferKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCommandFillBufferKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCommandFillImageKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clCommandNDRangeKernelKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clUpdateMutableCommandsKHR);
+        CL_HPP_INIT_CL_EXT_FCN_PTR_(clGetMutableCommandInfoKHR);
+#endif
+        if ((clCreateCommandBufferKHR        == nullptr) ||
+            (clFinalizeCommandBufferKHR      == nullptr) ||
+            (clRetainCommandBufferKHR        == nullptr) ||
+            (clReleaseCommandBufferKHR       == nullptr) ||
+            (clGetCommandBufferInfoKHR       == nullptr) ||
+            (clEnqueueCommandBufferKHR       == nullptr) ||
+            (clCommandBarrierWithWaitListKHR == nullptr) ||
+            (clCommandCopyBufferKHR          == nullptr) ||
+            (clCommandCopyBufferRectKHR      == nullptr) ||
+            (clCommandCopyBufferToImageKHR   == nullptr) ||
+            (clCommandCopyImageKHR           == nullptr) ||
+            (clCommandCopyImageToBufferKHR   == nullptr) ||
+            (clCommandFillBufferKHR          == nullptr) ||
+            (clCommandFillImageKHR           == nullptr) ||
+            (clCommandNDRangeKernelKHR       == nullptr) ||
+            (clUpdateMutableCommandsKHR      == nullptr) ||
+            (clGetMutableCommandInfoKHR      == nullptr))
+        {
+            ext_init_error_ = CL_INVALID_OPERATION;
+        }
+    }
+}; // CommandBufferKhr
+
+CL_HPP_DEFINE_STATIC_MEMBER_ std::once_flag CommandBufferKhr::ext_init_;
+CL_HPP_DEFINE_STATIC_MEMBER_ cl_int CommandBufferKhr::ext_init_error_ = CL_SUCCESS;
+
+/*! \class MutableCommandKhr
+ * \brief MutableCommandKhr interface for cl_mutable_command_khr.
+ */
+class MutableCommandKhr : public detail::Wrapper<cl_mutable_command_khr>
+{
+public:
+    //! \brief Default constructor - initializes to nullptr.
+    MutableCommandKhr() : detail::Wrapper<cl_type>() { }
+
+    explicit MutableCommandKhr(const cl_mutable_command_khr& mutableCommandKhr, bool retainObject = false) :
+        detail::Wrapper<cl_type>(mutableCommandKhr, retainObject) { }
+
+    MutableCommandKhr& operator=(const cl_mutable_command_khr& rhs)
+    {
+        detail::Wrapper<cl_type>::operator=(rhs);
+        return *this;
+    }
```

Comment:
Why do you need such an operator?

Reply:
For any scenario where object type cl_mutable_command_khr will be assigned to object class MutableCommandKhr.
Do you think is  this unreal case ?

Reply:
No, it is. But why the current code does not get one synthesized automatically?
You have already a `MutableCommandKhr` constructor from a `const cl_mutable_command_khr&`, so I guess the default `MutableCommandKhr::operator=()` would kick in.
Just some naive question since I do not have the big picture.

Reply:
No, it doesn't. This copy constructor will do this job if it won't have explicit specifier.

Reply:
Then perhaps this is a case where an implicit constructor makes sense instead of adding a lot of operators trying to mimic it?